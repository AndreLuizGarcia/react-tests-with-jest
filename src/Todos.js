import React, { Component } from 'react'
import { connect } from 'react-redux'

class Todos extends Component {
  render(){
    return(
      <ul className="Todos">
        { this.props.props.todos.map(todo => (
          <li key={todo.id}>{todo.text}</li>
        ))}
      </ul>
    )
  }
}

const mapStateToProps = state => ({
  todos: state.todos
})

export default connect(mapStateToProps)(Todos)  //connect para conectar nosso componente no redux