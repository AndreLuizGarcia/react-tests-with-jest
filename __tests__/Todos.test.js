import React from  'react'
import Todos from '../src/Todos'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'

const mockStore = configureStore()

const initialState = {
  todos: [
    { id: 0, text: 'Fazer cafÃ©'},
    { id: 1, text: 'Acessar o slack'},
    { id: 2, text: 'Ver RocketLive'}
  ]
}

const store = mockStore(initialState)

describe('Testing Todos Component', () => {
  it('render correctly', () => {
    // const wrapper = shallow(
    //   <Provider store={store}>
    //     <Todos />
    //   </Provider>
    // ).dive({context: {store}})     //worked

    const wrapper = shallow(
      <Todos store={store} />,
    ).dive();     //workedToo

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})