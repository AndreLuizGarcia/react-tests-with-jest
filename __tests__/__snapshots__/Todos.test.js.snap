// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Todos Component render correctly 1`] = `
<Todos
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  todos={
    Array [
      Object {
        "id": 0,
        "text": "Fazer cafÃ©",
      },
      Object {
        "id": 1,
        "text": "Acessar o slack",
      },
      Object {
        "id": 2,
        "text": "Ver RocketLive",
      },
    ]
  }
/>
`;
